src/main/scala/scorex/core/network/peer/PeerInfo.scala

```diff
@@ -2,19 +2,29 @@ package scorex.core.network.peer
 
 import java.net.InetSocketAddress
 
-import scorex.core.network.{ConnectionType, PeerFeature}
+import scorex.core.app.Version
+import scorex.core.network.{ConnectionType, Handshake, PeerData}
 
-case class PeerInfo(lastSeen: Long,
-                    declaredAddress: Option[InetSocketAddress],
-                    nodeName: Option[String] = None,
-                    connectionType: Option[ConnectionType] = None,
-                    features: Seq[PeerFeature] = Seq()) {
+/**
+  * Information about peer to be stored in PeerDatabase
+  *
+  * @param peerData       - general information about the peer
+  * @param lastSeen       - timestamp when this peer was last seen in the network
+  * @param connectionType - type of connection (Incoming/Outgoing) established to this peer if any
+  */
+case class PeerInfo(peerData: PeerData,
+                    lastSeen: Long,
+                    connectionType: Option[ConnectionType] = None) {
 
-  lazy val reachablePeer: Boolean = {
-    declaredAddress.isDefined || localAddress.isDefined
-  }
+  lazy val handshake: Handshake = Handshake(peerData, lastSeen)
 
-  lazy val localAddress: Option[InetSocketAddress] = {
-    features.collectFirst { case LocalAddressPeerFeature(addr) => addr }
-  }
 }
+
+object PeerInfo {
+
+  def fromAddress(address: InetSocketAddress): PeerInfo = {
```

Comment:
It's not clear that this method should be used for peers from config only. It would be better to make it obvious.